syntax = "proto3";

message PartyColumn {
  string name = 1;
  uint32 owner = 2;
  bool float_type = 3;
}

message PartyAddress {
  string ip_next = 1;
  string ip_prev = 2;
  uint32 port_next = 3;
  uint32 port_prev = 4;
}

enum TaskStatus {
  TASK_RUNNING = 0;
  TASK_DONE = 1;
  TASK_FAILED = 2;
  TASK_NOT_FOUND = 3;
}

message MPCExpressRequest {
  string jobid = 1;
  repeated PartyColumn columns = 2;
  optional uint32 local_partyid = 3;
  optional string expr = 4;
  optional PartyAddress addr = 5;
  optional string output_filepath = 6;
  optional string input_filepath = 7;
}

message MPCExpressResponse {
  string jobid = 1;
  string message = 2;
  optional TaskStatus status = 3;
}

service MPCExpressTask {
  rpc TaskStart(MPCExpressRequest) returns (MPCExpressResponse);
  rpc TaskStop(MPCExpressRequest) returns (MPCExpressResponse);
  rpc TaskStatus(MPCExpressRequest) returns (MPCExpressResponse);
}
